front/lib/api/assistant/agent_usage.ts

```diff
@@ -174,20 +179,46 @@ async function populateUsageIfNeeded({
   }
 }
 
-// Consider moving this to Redis if it's really slow.
+// We don't need real-time accuracy on the number of members
+// that added the agent in their list.
+// Best-effort, we cache the result for a day.
 async function getAgentInListCount(
   auth: Authenticator,
+  redis: Awaited<ReturnType<typeof redisClient>>,
+  workspaceId: string,
   agentConfiguration: LightAgentConfigurationType
 ) {
-  if (agentConfiguration.scope === "published") {
-    return getUsersWithAgentInListCount(auth, agentConfiguration.sId);
-  }
+  const { agentInListCountKey } = _getKeys({
+    agentConfigurationId: agentConfiguration.sId,
+    workspaceId,
+  });
 
-  if (agentConfiguration.scope === "workspace") {
-    return getMembersCount(auth, { activeOnly: true });
+  const cachedCount = await redis.get(agentInListCountKey);
+  if (cachedCount !== null) {
+    return parseInt(cachedCount, 10);
   }
 
-  return 0;
+  const calculateAndCacheCount = async (countPromise: Promise<number>) => {
+    const count = await countPromise;
```

Comment:
would it make sense to do that with the existing framework (i.e. in `populateRedisIfNeeded` like other counts)?

Reply:
The `populateRedisIfNeeded` will only get called if all the keys are missing in Redis. Today, on every agent's messages or user's mentions, we postpone the key expirations in Redis. It means that for some agents we basically never re-compute from the DB, which is not our case here. Given that this new logic is different than the existing one I'd encourage us to avoid cluttering the existing code.